<!--REQUEST_ID=042c2cdd-ba67-4cf9-a02d-ae98e63d7b8e-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48108&attempt_id=cd9b14f7-0ae0-4031-8d72-3bbb66b9af5b)

Attempted: teleconsole

<details><summary>Partial log (click to expand)</summary><p>

```
# github.com/gravitational/teleport/vendor/github.com/kr/pty
go/src/github.com/gravitational/teleport/vendor/github.com/kr/pty/pty_linux.go:34:8: undefined: _C_uint
go/src/github.com/gravitational/teleport/vendor/github.com/kr/pty/pty_linux.go:43:8: undefined: _C_int
# github.com/gravitational/teleport/vendor/github.com/boltdb/bolt
go/src/github.com/gravitational/teleport/vendor/github.com/boltdb/bolt/db.go:85:13: undefined: maxMapSize
# github.com/gravitational/teleport/vendor/github.com/kr/pty
go/src/github.com/gravitational/teleport/vendor/github.com/kr/pty/pty_linux.go:34:8: undefined: _C_uint
go/src/github.com/gravitational/teleport/vendor/github.com/kr/pty/pty_linux.go:43:8: undefined: _C_int
builder for '/nix/store/akq4gyfv3v83pm8rd1sb6za7malsgafm-teleconsole-0.4.0.drv' failed with exit code 5
error: build of '/nix/store/akq4gyfv3v83pm8rd1sb6za7malsgafm-teleconsole-0.4.0.drv' failed
```
</p></details>

--- a/pkgs/tools/misc/teleconsole/default.nix
+++ b/pkgs/tools/misc/teleconsole/default.nix
@@ -1,0 +1,34 @@
+{ stdenv, buildGoPackage, fetchFromGitHub }:
+
+buildGoPackage rec {
+  name = "teleconsole-${version}";
+  version = "0.4.0";
+
+  goPackagePath = "github.com/gravitational/teleconsole";
+
+  src = fetchFromGitHub {
+    owner = "gravitational";
+    repo = "teleconsole";
+    rev = version;
+    sha256 = "01552422n0bj1iaaw6pvg9l1qr66r69sdsngxbcdjn1xh3mj74sm";
+  };
+
+  goDeps = ./deps.nix;
+
+  CGO_ENABLED = 1;
+  buildFlags = "-ldflags";
+
+  meta = with stdenv.lib; {
+    homepage = "https://www.teleconsole.com/";
+    description = "Share your terminal session with people you trust";
+    license = licenses.asl20;
+    platforms = platforms.all;
+    # Builds for Aarch64 not possible in the current release due to
+    # incompatibilities further up the dependency chain.
+    # See:
+    #  - https://github.com/gravitational/teleport/issues/679
+    #  - https://github.com/kr/pty/issues/27
+    broken = stdenv.isAarch64;
+    maintainers = [ maintainers.kimburgess ];
+  };
+}

--- a/pkgs/tools/misc/teleconsole/deps.nix
+++ b/pkgs/tools/misc/teleconsole/deps.nix
@@ -1,0 +1,13 @@
+[
+    # Teleport v2.0.0-alpha.4 required for build.
+    # See https://github.com/gravitational/teleconsole/blob/09591f227c2a8df4c68af8bc4adfadfc596f4ed2/Makefile#L8
+    {
+      goPackagePath = "github.com/gravitational/teleport";
+      fetch = {
+        type = "git";
+        url = "https://github.com/gravitational/teleport";
+        rev = "2cb40abd8ea8fb2915304ea4888b5b9f3e5bc223";
+        sha256 = "1xw3bfnjbj88x465snwwzn4bmpmzmsrq9r0pkj388qwvfrclgnfk";
+      };
+    }
+]

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -5550,6 +5550,8 @@ with pkgs;
 
   teamviewer = libsForQt5.callPackage ../applications/networking/remote/teamviewer { };
 
+  teleconsole = callPackage ../tools/misc/teleconsole { };
+
   telegraf = callPackage ../servers/monitoring/telegraf { };
 
   teleport = callPackage ../servers/teleport {};

